init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceInvoker.java

```diff
@@ -16,32 +13,22 @@
 import java.util.List;
 
 @Service
-public class RoleServiceInvoker implements ServiceInvoker {
+public class RoleServiceInvoker extends ServiceInvoker {
     private static final Logger LOGGER = LoggerFactory.getLogger(RoleServiceInvoker.class);
 
     @Value("${role.service}")
     private String roleService;
 
-    private final WebClient webClient;
-
-    public RoleServiceInvoker() {
-        this.webClient = WebClient.create(roleService);
-    }
-    public RoleServiceInvoker(WebClient webClient) {
-        this.webClient = webClient;
-    }
+    public RoleServiceInvoker() { }
+    public RoleServiceInvoker(WebClient webClient) { super(webClient); }
 
     @Override
-    public Role send(Entity role) throws DependentServiceException {
+    protected String create(Entity role) throws DependentServiceException {
 
+        String uri = roleService + "/roles";
```

Comment:
maybe create endpoint api refs in the properties?

Reply:
Thanks Darren, I've made this change. Had to use a wildcard in the props file for this one /practitioner/%s/roles but spring property injection seems to handle that ok.